web/client/test-resources/wmc/context.wmc

```diff
@@ -23,8 +23,23 @@
         <ms:Group id="groupid3" title="Group3" expanded="true"/>
       </ms:GroupList>
       <ms:CatalogServices xmlns:ms="http://geo-solutions.it/mapstore/context" selectedService="testservice">
-        <ms:Service url="https://testserver/csw" type="csw" title="Demo CSW Service" autoload="1" oldService="old_service" serviceName="Demo CSW Service"/>
-        <ms:Service url="https://testserver/testservice/csw" type="csw" title="testservice" autoload="0" showAdvancedSettings="1" showTemplate="1" hideThumbnail="0" metadataTemplate="&lt;p>${description}&lt;/p>" format="image/jpeg" serviceName="testservices"/>
+        <ms:Service serviceName="Demo CSW Service">
+          <ms:Attribute name="url" type="string">https://testserver/csw</ms:Attribute>
+          <ms:Attribute name="type" type="string">csw</ms:Attribute>
+          <ms:Attribute name="title" type="string">Demo CSW Service</ms:Attribute>
+          <ms:Attribute name="autoload" type="boolean">true</ms:Attribute>
+          <ms:Attribute name="oldService" type="string">old_service</ms:Attribute>
+        </ms:Service>
+        <ms:Service serviceName="testservices">
+          <ms:Attribute name="url" type="string">https://testserver/testservice/csw</ms:Attribute>
+          <ms:Attribute name="type" type="string">csw</ms:Attribute>
+          <ms:Attribute name="title" type="string">testservice</ms:Attribute>
+          <ms:Attribute name="autoload" type="boolean">false</ms:Attribute>
+          <ms:Attribute name="showAdvancedSettings" type="boolean">true</ms:Attribute>
+          <ms:Attribute name="showTemplate" type="boolean">true</ms:Attribute>
+          <ms:Attribute name="metadataTemplate" type="string">&lt;p>${description}&lt;/p></ms:Attribute>
```

Comment:
you should use `<![CDATA[<p>${description}</p>]]>` for strings, so you can avoid to take care about xml encoding, ant it's useful also for JSON.
Please add also a test for JSON object (options attribute).

Reply:
>  I seed you needed to add it, but it can become problematic quickly

What do you mean? All needed characters are escaped by default.

Reply:
Yes but are you sure you're covering all the use cases?
Expecially for unicode chars and so on. Please try to export and re-import also a metadata template (you can use the UI in new catalog ) with special chars like àèìòù. CDATA is more secure for unicode.

Reply:
Only control chars need to be escaped, these are well defined.